handler/live.go

```diff
@@ -23,42 +23,32 @@ var (
 )
 
 func (h *Handler) live(w http.ResponseWriter, r *http.Request) {
-	// Parse username from request query
-	username := r.URL.Query().Get("username")
-	if username == "" {
-		w.WriteHeader(http.StatusBadRequest)
-		w.Write([]byte("invalid username"))
-	}
-
-	// Parse roomID from request query
-	roomID, err := uuid.Parse(r.URL.Query().Get("room_id"))
-	if err != nil {
-		w.WriteHeader(http.StatusBadRequest)
-		w.Write([]byte("invalid room id"))
+	// Parse token from request query
+	token := r.URL.Query().Get("token")
+	username, gameID, ok := h.srv.GetTokenPayload(token)
+	if !ok {
+		resp.Error(w, errs.B().Code(errs.InvalidArgument).Msg("invalid token").Err())
 		return
 	}
 
 	// Get room from Hub
 	Hub.mu.Lock()
-	defer Hub.mu.Unlock()
-	room, ok := Hub.rooms[roomID]
+	room, ok := Hub.rooms[gameID]
+	Hub.mu.Unlock()
 	if !ok {
```

Comment:
RWMutex will be better to use for the Hub because most of the times, when we need a room, we have a lot of reads, and it doesn't make sense for us to lock other readers too.